transport/httprp/server.go

```diff
@@ -0,0 +1,62 @@
+package httprp
+
+import (
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+
+	"golang.org/x/net/context"
+)
+
+// RequestFunc may take information from an HTTP request and put it into a
+// request context. BeforeFuncs are executed prior to invoking the
+// endpoint.
+type RequestFunc func(context.Context, *http.Request) context.Context
+
+// Server wraps an endpoint and implements http.Handler.
```

Comment:
I guess it's not wrapping an endpoint anymore, but rather a proxying http.Handler, right?